highfidelity/hifi
# Updating android toolchain for QML

* Support Gradle functionality similar to `androiddeployqt`
  * Use qmlimportscanner to find all the required system QML files and import them into the assets folder
  * Use a predefined list of Qt dependencies to parse the Qt android dependency XML files and copy over the required plugins in their 'garbled' form
  * Generate a libs.xml file that includes the list of dependency files and plugins with a mapping of the garbled names to the relative paths
  * Update the android manifest to expose the libs.xml data to the activity loader
  * Integrate a modified version of QtActivityLoader to parse the libs.xml and do appropriate unpacking and loading of the assets
* Update the primary app to use similar initialization logic to Qt examples
  * Include a `main` function that creates a Qt application the same way it would for desktop

## Testing

This PR makes minimal changes to the code shared between desktop and Android.  The bulk of the changes are additions of `#if !defined(Q_OS_ANDROID)` blocks demarking code that is not appropriate for or compatible with Android.  

* Verify that offscreen surface rendering still works
  * Web Entities, Tablet UI, Toolbar UI
* Verify the same display plugins are listed for this PR as for the master build
* Basic smoke test
